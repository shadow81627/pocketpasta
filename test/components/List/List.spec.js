import Component from '@/components/List/List';
import { shallowMount, createLocalVue, RouterLinkStub } from '@vue/test-utils';
import Vuex from 'vuex';
import BootstrapVuePlugin from 'bootstrap-vue';

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(BootstrapVuePlugin);

describe('List', () => {
  const factory = () =>
    shallowMount(Component, {
      localVue,
      stubs: {
        NuxtLink: RouterLinkStub,
      },
    });

  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
