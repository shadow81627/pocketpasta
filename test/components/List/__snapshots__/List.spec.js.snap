// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders properly 1`] = `
"<div>
  <div class=\\"row\\">
    <div class=\\"col-auto mr-auto\\">
      <h1></h1>
    </div>
    <div class=\\"col-auto ml-auto my-auto\\">
      <view-switcher-stub layouts=\\"[object Object],[object Object]\\"></view-switcher-stub>
    </div>
  </div>
  <div class=\\"row\\">
    <div class=\\"col-12\\">
      <div class=\\"card-layout\\"></div>
    </div>
  </div>
</div>"
`;
